pkg/util/validation/validate.go

```diff
@@ -65,8 +73,16 @@ func (cfg *Overrides) ValidateSample(userID string, metricName string, s client.
 	return nil
 }
 
+// LabelValidationConfig helps with getting required config to validate labels
+type LabelValidationConfig interface {
```

Comment:
`*Config` suggests a struct and not an interface. Can we rename it to `LabelValidator`. Similarly for `SampleValidator`